### PR TITLE
[Issue 135]: Add swagger specs for jupyter-websocket mode

### PR DESCRIPTION
**Pull request by Lull3rSkat3r**
(c) Copyright IBM Corp. 2016

### CODE REVIEW

File: kernel_gateway/mixins.py [hunk 1]
```diff
@@ -56,6 +56,8 @@ def prepare(self):
         server_token = self.settings.get('kg_auth_token')
         if server_token:
             client_token = self.request.headers.get('Authorization')
+            print('Client token is %s' % client_token)
+            print('Server token is %s' % server_token)
```
Comment (parente) [inline kernel_gateway/mixins.py]: Remove prints.

File: kernel_gateway/services/api/handlers.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+class APIYamlHandler(BaseSpecHandler):
+    """Exposes a YAML swagger specification"""
+    @staticmethod
+    def get_resource_metadata():
+        return ('swagger.yaml', 'text/x-yaml')
+
+
+default_handlers = [
+    ('/api/{}'.format(SpecJsonHandler.get_resource_metadata()[0]), SpecJsonHandler),
+    ('/api/{}'.format(APIYamlHandler.get_resource_metadata()[0]), APIYamlHandler)
```
Comment (parente) [inline kernel_gateway/services/api/handlers.py]: Double check: will this work with a custom base_url?

File: kernel_gateway/services/api/swagger.json [hunk 3]
```diff
@@ -0,0 +1,564 @@
...
+                        }
+                    }
+                },
+                "kernel": {
+                    "$ref": "#/definitions/Kernel"
+                }
+            }
+        }
+    }
+}
```
Comment (parente) [inline kernel_gateway/services/api/swagger.json]: How is the JSON produced so that when the YML is updated, the JSON can be updated too?

Comment (parente) [inline kernel_gateway/services/api/swagger.json]: Following up on this: we should update the documentation for contributors about when to update the Swagger spec and how to regenerate the JSON from the YML.

Comment (Lull3rSkat3r) [inline kernel_gateway/services/api/swagger.json]: I have been using the [swagger editor](http://editor.swagger.io) and just exporting both from there. I can update the docs for development on how to edit and update these files.

File: kernel_gateway/services/api/handlers.py (hunk 2 above)
Comment (Lull3rSkat3r) [inline kernel_gateway/services/api/handlers.py]: Yes, tested locally and works. These endpoints get added to the list of urls which get prepended with the base url in [`gatewayapp.py`](https://github.com/jupyter/kernel_gateway/blob/master/kernel_gateway/gatewayapp.py#L359)

File: README.md [hunk 4]
```diff
@@ -102,3 +102,12 @@ straightforward using these steps:
       route `/api/kernels` is not enabled by default for greater security. See
       the `--KernelGatewayApp.list_kernels` parameter documentation if you
       would like to enable the `/api/kernels` route.)
+
+### Swagger Documentation
```
Comment (Lull3rSkat3r) [inline README.md]: @parente, here are the docs for updating the swagger specs. Let me know if you would like more.

Comment (parente) [inline README.md]: I think these need to live in the proper docs now. The README is limited to the getting started info. I suggest moving it to https://github.com/jupyter/kernel_gateway/blob/master/docs/source/devinstall.md for now.

File: docs/source/websocket-mode.md [hunk 5]
```diff
@@ -1,21 +1,7 @@
...
-Discounting features of the kernel gateway (e.g., token auth), the behavior
-of these resources is equivalent to that found in the Jupyter Notebook server.
-The kernel gateway simply imports and extends the handler classes from
-the Jupyter Notebook.
\ No newline at end of file
+The `KernelGatewayApp.api` command line argument defaults to
+`jupyter-websocket`. The API exposed in this mode can be viewed
+in the [swagger ui.](http://petstore.swagger.io) by entering
+the link to the [swagger spec](https://raw.githubusercontent.com/jupyter/kernel_gateway/master/kernel_gateway/services/api/swagger.json)
+of kernel gateway.
```
Comment (parente) [inline docs/source/websocket-mode.md]: Can you also include a direct link to the YAML on master? It's pretty readable. Don't have to use the UI always.

**Comment by parente** [comment]
Nevermind my latest comments. I'll do any doc cleanup in #143. Let's :shipit: !

**Pull request merged**